---
title: "HERS2022_Final_R_Code"
author: "Veronica Iriart"
date: "2025-03-06"
output: html_document
---

## Load Packages
```{r}
library(ggplot2) #make plots
library(tidyr) # has function gather and other data cleaning functions
library(Rmisc) # summarySE function
library(dplyr) # data filtering and cleaning 
library(ape) # phylogenetic analysis and tree building
library(coxme) # cox proportional hazards models with mixed effects for time-to-event analysis
library(survival) # time-to-event analysis functions
library(car) # Anova type III function
library(AICcmodavg) # model fit and summaries
library(pammtools) # has function geom_stepribbon for plotting time to event results
library(ggpubr)# has ggerrorplot for data visualizations
library(lme4) # linear mixed effects models
library(DHARMa) # model diagnostics
library(emmeans) # emmeans calculations
library(geiger) # phylogenetic analysis
library(textshape) # function column_to_rownames
library(phyr) # phylogeny-corrected linear models
```

## Phylogeny of rhizobial strains
```{r}
#### This chunk plots the best-scoring maximum likelihood tree from RAxML (Ranodmized Axelerated Maximum Likelihood) program analysis performed on the remote server.
rhizobia.tree <- read.tree(file = "RAxML_bestTree.HERS2022_tree.txt")

plot.phylo(rhizobia.tree) # see the tree

### Rename tip labels with strain IDs
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="genomic.gbff"] <- "strain_14479"
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="scaffolds_2788_10.gbff"] <- "strain_2220"
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="scaffolds_2788_17.gbff"] <- "strain_4386"
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="scaffolds_2788_5.gbff"] <- "strain_2141"
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="scaffolds_2788_8.gbff"] <- "strain_2214"
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="scaffolds_2788_4.gbff"] <- "strain_2087"
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="scaffolds_2788_12.gbff"] <- "strain_2316"
rhizobia.tree$tip.label[rhizobia.tree$tip.label=="scaffolds_2788_3.gbff"] <- "strain_2063"

plot(rhizobia.tree) # final tree
```

## Time until nodulation analysis
```{r}
#### This chunk analyzes time until nodulation data from the microcosm study

### Prep data
plant_dat <- read.csv("HERS2022_Microcosm_Exp_Plant_Data_03-07-25.csv") # load master dataset from microcosm study

## Filter out uninoculated plants (2 uninoc. plants counted as having 1-2 nodules due to human error/limitations with counting from photographs), missing data, and plants noted to have severe symptoms from a fungal pathogen
nod_in_dat <- plant_dat %>% filter(strain_ID != "uninoculated") %>% 
  filter(WK4_num_nodules != "NA") %>%
  filter(WK4_fungal_pathogen_severe != "Y")

## Make a column for nodule initiation (first_nodule_timing) based on when nodules were first recorded to be present (week 1, 2, 3, or 4)

nod_in_dat$WK4_nodule_presence <- 2

for(i in 1:nrow(nod_in_dat)){
  if(nod_in_dat[i,7] != 0) {
    nod_in_dat[i,15] <- 1
  }
  else{
    nod_in_dat[i,15] = 0
  }
  i <- i + 1
}

nod_in_dat <- nod_in_dat %>%
  mutate(nodule_presence_sum = WK1_nodule_presence + WK2_nodule_presence + WK3_nodule_presence + WK4_nodule_presence)

nod_in_dat$first_nodule_timing <- 0

for(i in 1:nrow(nod_in_dat)) {
  if(nod_in_dat[1,16] == 4){
    nod_in_dat[i,17] <- 1
  }
  if(nod_in_dat[i,16] == 3){
    nod_in_dat[i,17] <- 2
  }
  if(nod_in_dat[i,16] == 2){
    nod_in_dat[i,17] <- 3
  }
  if(nod_in_dat[i,16] == 1){
    nod_in_dat[i,17] <- 4
  }
  if(nod_in_dat[i,16] == 0){
    nod_in_dat[i,17] <- 4
  }
  i <- i + 1
}

## Make a column to indicate whether first_nodule_timing was censored, i.e. whether plants had nodules present on Week 4 (censored = 1) or did not have nodules by Week 4 (censored = 0; meaning no event occurred)
nod_in_dat$censored <- 1
for(i in 1:nrow(nod_in_dat)){
  if(nod_in_dat[i,17] == 4 & nod_in_dat[i,16] == 0){
    nod_in_dat[i,18] <- 0
  }
}

### Model selection

## Ensure response and explanatory variables and covariates are saved in proper format

nod_in_dat$Tray <- as.factor(nod_in_dat$Tray)
nod_in_dat$temporal_block <- as.factor(nod_in_dat$temporal_block)
nod_in_dat$plant_genotype <- as.factor(nod_in_dat$plant_genotype)
nod_in_dat$dicamba_trt <- as.factor(nod_in_dat$dicamba_trt)
nod_in_dat$strain_ID <- as.factor(nod_in_dat$strain_ID)
nod_in_dat$WK4_fungal_pathogen_minor <- as.factor(nod_in_dat$WK4_fungal_pathogen_minor)

## Model Selection. We're choosing from mixed-effects cox proportional hazards models because these models evaluate the effect of random and fixed effect factors on the time to an event. In this case the event of interest is nodulation.

nod_in_coxme1 <- coxme(Surv(first_nodule_timing, censored) ~ dicamba_trt*strain_ID*plant_genotype + (1|Tray) + temporal_block + WK4_fungal_pathogen_minor, data = nod_in_dat) 
AIC(nod_in_coxme1) #2714.6

nod_in_coxme2 <- coxme(Surv(first_nodule_timing, censored) ~ dicamba_trt*strain_ID*plant_genotype + (1|Tray) + temporal_block, data = nod_in_dat) 
AIC(nod_in_coxme2) #2713.1

nod_in_coxme3 <- coxme(Surv(first_nodule_timing, censored) ~ dicamba_trt*strain_ID + dicamba_trt*plant_genotype + strain_ID*plant_genotype + (1|Tray) + temporal_block, data = nod_in_dat)
AIC(nod_in_coxme3) #2707.5

nod_in_coxme4 <- coxme(Surv(first_nodule_timing, censored) ~ dicamba_trt*strain_ID + dicamba_trt*plant_genotype + (1|Tray) + temporal_block, data = nod_in_dat)
AIC(nod_in_coxme4) #2700.6 -> best-fit model 

nod_in_models_all <- list(nod_in_coxme1, nod_in_coxme2, nod_in_coxme3, nod_in_coxme4)
aictab(nod_in_models_all)

### Model validation
summary(nod_in_coxme4)
test.ph <- cox.zph(nod_in_coxme4)
test.ph # the test is not statistically sig for each of the covariates, and global test is also not statistically significant, thus we can assume the proportional hazards. The proportional hazards assumption is the assumption that the survival curves for two or more strata (i.e. factors) must have hazard functions that are proportional over time.

### Hypothesis test
Anova(nod_in_coxme4, type = 3) # sig effects of dicamba trt and strain, no interactions

### Visualize results

## Plot main effect of dicamba
nod_in_fit <- survfit(Surv(first_nodule_timing, censored) ~ dicamba_trt, data = nod_in_dat)
print(nod_in_fit, print.rmean=TRUE) # mean and se's of timing of nodulation of each dicamba trt

nod_in_fit_dat <- with(nod_in_fit, data.frame(time, n.event, surv, std.err, lower, upper))
nod_in_fit_dat$dicamba_trt <- c("0", "0", "0", "0", "0.5", "0.5", "0.5", "0.5")
nod_in_fit_dat$surv2 <- 1 - nod_in_fit_dat$surv # calculate probability of nodulation (1-probability of not having a nodule)

nod_in_fit_dat$dicamba_trt2 <- "No Dicamba"
for(i in 1:nrow(nod_in_fit_dat)){
  if(nod_in_fit_dat[i,7]==0.5){
    nod_in_fit_dat[i,9] <- "Dicamba"
  }
}

nod_in_fit_dat$dicamba_trt2 <- factor(nod_in_fit_dat$dicamba_trt2, levels = c("No Dicamba", "Dicamba"))

nod_time_A <- ggplot(data = nod_in_fit_dat, aes(x = time, y = surv2, linetype = dicamba_trt2)) +
  geom_step(size = 2) +
  theme_classic() +
  geom_stepribbon(aes(ymin = surv2-std.err, ymax = surv2+std.err, linetype = dicamba_trt2), alpha = 0.25) +
  labs(y = "Probability of Nodulation", x = "Weeks Post-Inoculation", linetype = "Herbicide \nTreatment") +
  theme(axis.title = element_text(size = 24), axis.text = element_text(size = 22), legend.title = element_text(size = 20), legend.text = element_text(size = 16), legend.key.size = unit(1,'cm'))
nod_time_A

## Plot main effects of strains
nod_in_fit2 <- survfit(Surv(first_nodule_timing, censored) ~ strain_ID, data = nod_in_dat)
print(nod_in_fit2, print.rmean=TRUE) # mean and se's of timing of nodulation of each strain
nod_in_fit_dat2 <- with(nod_in_fit2, data.frame(time, n.event, surv, std.err, lower, upper))

nod_in_fit_dat2$strain_ID <- "strain"
nod_in_fit_dat2[1:4,7] <- "14479"
nod_in_fit_dat2[5:8,7] <- "2063"
nod_in_fit_dat2[9:12,7] <- "2087"
nod_in_fit_dat2[13:16,7] <- "2141"
nod_in_fit_dat2[17:20,7] <- "2214"
nod_in_fit_dat2[21:24,7] <- "2220"
nod_in_fit_dat2[25:28,7] <- "2316"
nod_in_fit_dat2[29:32,7] <- "4386"

nod_in_fit_dat2$surv2 <- 1 - nod_in_fit_dat2$surv # calculate probability of nodulation (1-probability of not having a nodule)


nod_timeB <- ggplot(data = nod_in_fit_dat2, aes(x = time, y = surv2)) +
  geom_step(size = 2) +
  theme_classic() +
  geom_stepribbon(aes(ymin = surv2-std.err, ymax = surv2+std.err), alpha = 0.25) +
  labs(y = "Probability of Nodulation", x = "Weeks Post-Inoculation") +
  theme(axis.title = element_text(size = 24), axis.text = element_text(size = 22),    strip.text = element_text(size = 16)) +
  facet_wrap(~strain_ID, ncol = 2)
nod_timeB 

```

## Nodule number analysis
```{r}
#### This chunk analyzes nodule number data from the microcosm study.

### Prep data

## Filter out uninoculated plants, inoculated plants that did not form nodules, missing data, and plants with severe symptoms from fungal pathogen contaminant
nod_num_dat <- plant_dat %>% filter(strain_ID != "uninoculated") %>% 
  filter(WK4_num_nodules != 0) %>%
  filter(WK4_fungal_pathogen_severe != "Y") %>%
  filter(WK4_num_nodules != "NA")

## Ensure variables are stored in correct class
nod_num_dat$dicamba_trt <- factor(nod_num_dat$dicamba_trt)
nod_num_dat$plant_genotype <- factor(nod_num_dat$plant_genotype) 
nod_num_dat$strain_ID <- factor(nod_num_dat$strain_ID) 
nod_num_dat$temporal_block <- factor(nod_num_dat$temporal_block) 
nod_num_dat$Tray <- factor(nod_num_dat$Tray)
nod_num_dat$WK4_fungal_pathogen_minor <- factor(nod_num_dat$WK4_fungal_pathogen_minor)
class(nod_num_dat$WK4_num_nodules) # integer
nod_num_dat$WK4_num_nodules <- as.numeric(nod_num_dat$WK4_num_nodules)
class(nod_num_dat$WK4_num_nodules) # numeric

### Data exploration

## Visualize raw data

nod_num_plot <- ggerrorplot(nod_num_dat, x="strain_ID", y="WK4_num_nodules",  desc_stat="mean_se", color="dicamba_trt", position=position_dodge(.3), xlab="strain", ylab="nodule number", add="jitter", add.params=list(color="dicamba_trt"), facet.by = "plant_genotype")
nod_num_plot + theme(axis.text.x = element_text(angle=90,hjust=1))

nod_num_plot2 <- ggerrorplot(nod_num_dat, x="strain_ID", y="WK4_num_nodules",  desc_stat="mean_se", color="plant_genotype", position=position_dodge(.3), xlab="strain", ylab="nodule number", add="jitter", add.params=list(color="plant_genotype"), facet.by = "dicamba_trt")
nod_num_plot2 + theme(axis.text.x = element_text(angle=90,hjust=1))

## Examine distribution of data
plot(density(nod_num_dat$WK4_num_nodules)) # very right skewed
plot(density(sqrt(nod_num_dat$WK4_num_nodules))) # slightly right skewed
plot(density(log(nod_num_dat$WK4_num_nodules))) # more normal

### Model selection. We're choosing from mixed effects linear models to analyze log-transformed nodule number data because these models are appropriate for the data distribution (normal) and can evaluate random and fixed effect factors. 

nod_num_lmer_mod1 <- lmer(log(WK4_num_nodules) ~ dicamba_trt*plant_genotype*strain_ID + (1|Tray) + WK4_fungal_pathogen_minor + temporal_block, data = nod_num_dat, REML = FALSE)
AIC(nod_num_lmer_mod1) #547.0

nod_num_lmer_mod2 <- lmer(log(WK4_num_nodules) ~ dicamba_trt*plant_genotype*strain_ID + (1|Tray) + temporal_block, data = nod_num_dat, REML = FALSE)
AIC(nod_num_lmer_mod2) #554.9

nod_num_lmer_mod3 <- lmer(log(WK4_num_nodules) ~ dicamba_trt*plant_genotype + dicamba_trt*strain_ID + plant_genotype*strain_ID + (1|Tray) + WK4_fungal_pathogen_minor + temporal_block, data = nod_num_dat, REML = FALSE)
AIC(nod_num_lmer_mod3) #543.7 <- best-fit model

nod_num_lmer_mod4 <- lmer(log(WK4_num_nodules) ~ dicamba_trt*plant_genotype + dicamba_trt*strain_ID + (1|Tray) + WK4_fungal_pathogen_minor + temporal_block, data = nod_num_dat, REML = FALSE)
AIC(nod_num_lmer_mod4) #546.3

nod_num_models_all <- list(nod_num_lmer_mod1, nod_num_lmer_mod2, nod_num_lmer_mod3, nod_num_lmer_mod4)
aictab(nod_num_models_all)

### Model Validation
simulationOutput <- simulateResiduals(fittedModel = nod_num_lmer_mod3, plot = F)
plot(simulationOutput) # passes
plot(histogram(residuals(nod_num_lmer_mod3))) # residuals are normally distributed

### Test Hypotheses
Anova(nod_num_lmer_mod3, type = 3) # sig effects of strain, fungal pathogen, temporal black, dicamba x strain, and plant x strain

### Post-hoc tests of interactions

## Calculate estimated marginal means from best-performing model
nod_num_emmeans <- emmeans(nod_num_lmer_mod3, pairwise ~ dicamba_trt|strain_ID)

## Examine herbicide x rhizobial strain effects
nod_num_contrast_dat <- as.data.frame(nod_num_emmeans$contrasts)
summary(nod_num_emmeans, by = NULL, adjust = "dunnettx")# Dunnet p-value adjustments, these results are reported in Supplementary Table S3 (Part A)

##  Examine plant genotype x rhizobial strain effects
nod_num_emmeans_plant <- emmeans(nod_num_lmer_mod3, pairwise ~ plant_genotype|strain_ID)
nod_num_emmeans_plant
nod_num_plant_contrast_dat <- as.data.frame(nod_num_emmeans_plant$contrasts)
summary(nod_num_emmeans_plant, by = NULL, adjust = "dunnettx") # Dunnet p-value adjustments, these results are reported in Supplementary Table S4


### Visualize results

## Herbicide x strain results
nod_num_emmeans2 <- emmeans(nod_num_lmer_mod3, pairwise ~ dicamba_trt|strain_ID, type = "response", adjust = "Tukey") # calculate back-transformed nodule number emmeans for ease of interpretation when plotting
nod_num_emmeans2
nod_num_emmeans_dat <- as.data.frame(nod_num_emmeans2)
nod_num_emmeans_dat <- nod_num_emmeans_dat[1:16,]
nod_num_emmeans_dat$strain_ID2 <- c("14479", "14479", "2063", "2063", "2087", "2087","2141", "2141", "2214", "2214", "2220", "2220", "2316", "2316", "4386", "4386")

# Prep dataframe to generate plot

nod_num_emmeans_dat$dicamba_trt2 <- "No Dicamba"
for(i in 1:nrow(nod_num_emmeans_dat)){
  if(nod_num_emmeans_dat[i,2] == 0.5){
    nod_num_emmeans_dat[i,10] <- "Dicamba"
  }
}

nod_num_emmeans_dat$dicamba_trt2 <- factor(nod_num_emmeans_dat$dicamba_trt2, levels = c("No Dicamba", "Dicamba"))

# Plot results

c8 <- c("#40B0A6", "#f0e442", "skyblue1", "black", "#cc79a9", "#785ef0", "#d55e00", "darkgoldenrod2") # strain colors

nod_num_rxn_norm <- ggplot(nod_num_emmeans_dat, aes(x = dicamba_trt2, y = response, color = strain_ID2)) +
  geom_point(size = 3, position = position_dodge(width = 0.3)) +
  geom_errorbar(aes(ymin = response-SE, ymax = response + SE), width = 0, size = 1, position = position_dodge(width = 0.3)) +
  labs(x = "Herbicide Treatment", y = "Nodule No.", color = "Strain") +
  theme_bw() +
  theme(axis.title = element_text(size = 18), axis.text = element_text(size = 16), legend.title = element_text(size = 18), legend.text = element_text(size = 16)) +
  geom_line(aes(group = strain_ID2), position = position_dodge(width = 0.3), size = 1) +
    scale_color_manual(values = c8)
nod_num_rxn_norm

## Plant x strain results
nod_num_emmeans_plant2 <- emmeans(nod_num_lmer_mod3, pairwise ~ plant_genotype|strain_ID, type = "response", adjust = "Tukey") # back-transformed emmeans
nod_num_emmeans_plant_dat <- as.data.frame(nod_num_emmeans_plant2$emmeans)
nod_num_emmeans_plant_dat$strain_ID2 <- c("14479", "14479", "2063", "2063", "2087", "2087","2141", "2141", "2214", "2214", "2220", "2220", "2316", "2316", "4386", "4386")

# Plot results

nod_num_rxn_norm2 <- ggplot(nod_num_emmeans_plant_dat, aes(x = plant_genotype, y = response, color = strain_ID2)) +
  geom_point(size = 3, position = position_dodge(width = 0.3)) +
  geom_errorbar(aes(ymin = response-SE, ymax = response + SE), width = 0, size = 1, position = position_dodge(width = 0.3)) +
  labs(x = "Plant Genotype", y = "Nodule No.", color = "Strain") +
  theme_bw() +
  theme(axis.title = element_text(size = 18), axis.text = element_text(size = 16), legend.title = element_text(size = 18), legend.text = element_text(size = 16)) +
  geom_line(aes(group = strain_ID2), position = position_dodge(width = 0.3), size = 1) +
    scale_color_manual(values = c8)
nod_num_rxn_norm2

### Spearman correlation: are strain rankings of nodule number the same with or without dicamba?
nod_num_emmeans_dat_no_dicamba <- nod_num_emmeans_dat %>%
  filter(dicamba_trt == "0")
nod_num_emmeans_dat_dicamba <- nod_num_emmeans_dat %>%
  filter(dicamba_trt == "0.5")

cor.test(nod_num_emmeans_dat_no_dicamba$response, nod_num_emmeans_dat_dicamba$response, method = "spearman")
#r = -0.40, p = 0.33


```

## BNF analysis
```{r}
#### This chunk analyzes biological nitrogen fixation (BNF) data from the microcosm study which was estimated from foliar delta-15 N data. 

### Prep data
N_data <- plant_dat %>% filter(temporal_block == 1) # only submitted samples from first temporal block 

## Foliar N Isotope analysis was performed by an external lab (Washington State University [WSU]), therefore we have to first merge the dataset obtained from WSU with our dataset of data collected from the plant microcosm experiment to perform this analysis.

N_isotope_sample_selection <- read.csv("HERS2022_Microcosm_Exp_Isotope_Analysis_Sample_Selection_02-21-24.csv")
N_isotope_sample_selection <- N_isotope_sample_selection[1:164,1:2] 

N_data2 <- merge(N_data, N_isotope_sample_selection, by = "Plate_ID")
names(N_data2)[15] <- "Isotope.ID"

N_isotope_data <- read.csv("HERS2022_Microcosm_Exp_N_Isotope_Data_04-17-24.csv")

N_data3 <- merge(N_data2, N_isotope_data, by = "Isotope.ID")

names(N_data3)[19] <- "delta_15N" # this is the column with the data w are interested in, because it estimates rhizobial BNF activity

### Data Exploration

## Ensure explanatory variables are stored as the correct class

N_data3$Tray <- factor(N_data3$Tray)
N_data3$plant_genotype <- factor(N_data3$plant_genotype)
N_data3$strain_ID <- factor(N_data3$strain_ID)
N_data3$dicamba_trt <- factor(N_data3$dicamba_trt)
N_data3$WK4_fungal_pathogen_minor <- factor(N_data3$WK4_fungal_pathogen_minor)
class(N_data3$delta_15N) # numeric

## Visualize raw data

ggerrorplot(N_data3, x="strain_ID", y="delta_15N",  desc_stat="mean_se", color="dicamba_trt", position=position_dodge(.3), xlab="Rhizobial Strain", ylab="Leaf delta 15N (raw means)", add="jitter") +
  theme(axis.text = element_text(size = 16), axis.title = element_text(size=18)) +
  scale_color_manual(values = c("dodgerblue3", "salmon2")) +
  coord_flip()

## Examine mean delta 15 N values according to inoculation treatment. Compare means of plants inoculated with rhizobia vs. uninoculated. Uninoculated plants should have higher delta 15 N values than plants inoculated with rhizobia
inoc_N_dat <- summarySE(N_data3, measurevar = "delta_15N", groupvars = c("strain_ID"))
inoc_N_dat 

## Examine distribution of data
plot(density(N_data3$delta_15N)) #looks pretty normally distributed
plot(histogram(N_data3$delta_15N))

### Model selection. We're choosing from mixed effects linear models to analyze foliar delta-15 N data because these models are appropriate for the data distribution (normal) and can evaluate random and fixed effect factors. 

N_data4 <- N_data3 %>% filter(strain_ID != "uninoculated") # remove uninoculated plants from data as we are only interested in making comparisons among plants that interacted with rhizobia

N_model1 <- lmer(delta_15N ~ dicamba_trt*strain_ID*plant_genotype + (1|Tray), data = N_data4, REML = FALSE)
AIC(N_model1) #568.7

N_model2 <- lmer(delta_15N ~ dicamba_trt*strain_ID*plant_genotype + (1|Tray) + WK4_fungal_pathogen_minor, data = N_data4, REML = FALSE)
AIC(N_model2) #565.0

N_model3 <- lmer(delta_15N ~ dicamba_trt*strain_ID + dicamba_trt*plant_genotype + strain_ID*plant_genotype + (1|Tray) + WK4_fungal_pathogen_minor, REML= FALSE, data = N_data4)
AIC(N_model3) #559.2

N_model4 <- lmer(delta_15N ~ dicamba_trt*strain_ID + dicamba_trt*plant_genotype + (1|Tray) + WK4_fungal_pathogen_minor, REML = FALSE, data = N_data4)
AIC(N_model4) #549.6 <- best-fit model


N_models_all <- list(N_model1, N_model2, N_model3, N_model4)
aictab(N_models_all)

### Model Validation
simulationOutput <- simulateResiduals(fittedModel = N_model4, plot = F)
plot(simulationOutput) # passes
plot(histogram(residuals(N_model4))) # residuals are normally distributed

### Test hypotheses
Anova(N_model4, type = 3) # sig effects of strain, fungal pathogen and dic x strain 

### Post-hoc tests of interactions

## Calculate estimated marginal means from best-performing model
N_emmeans <- emmeans(N_model4, pairwise ~ dicamba_trt|strain_ID)

## Examine pairwise comparisons of dicamba effect by strain
N_emmeans_contrast_dat <- as.data.frame(N_emmeans$contrasts)
N_emmeans_contrast_dat
summary(N_emmeans, by = NULL, adjust = "dunnettx") # Dunnet p-value adjustments


### Visualize results

## Calculate estimated marginal means for foliar delta-15 N data from N_model3 (best model)

N_emmeans <- emmeans(N_model3, pairwise ~ dicamba_trt|strain_ID, type = "response", adjust = "Tukey")
N_emmeans_dat <- as.data.frame(N_emmeans$emmeans)

## Examine pairwise comparisons of dicamba effect by strain
N_emmeans_contrast_dat <- as.data.frame(N_emmeans$contrasts)
N_emmeans_contrast_dat
summary(N_emmeans, by = NULL, adjust = "dunnettx") # Dunnet p-value adjustments

## Prep dataframe to generate plot
N_emmeans_dat$strain_ID2 <- c("14479", "14479", "2063", "2063", "2087", "2087","2141", "2141", "2214", "2214", "2220", "2220", "2316", "2316", "4386", "4386")

N_emmeans_dat$emmean2 <- N_emmeans_dat$emmean*-1
N_emmeans_dat$se2 <- N_emmeans_dat$SE*-1

N_emmeans_dat$dicamba_trt2 <- "No Dicamba"
for(i in 1:nrow(N_emmeans_dat)){
  if(N_emmeans_dat[i,1] == 0.5){
   N_emmeans_dat[i,11] <- "Dicamba"
  }
}

N_emmeans_dat$dicamba_trt2 <- factor(N_emmeans_dat$dicamba_trt2, levels = c("No Dicamba", "Dicamba"))

## Plot results
N_rxn_norm <- ggplot(N_emmeans_dat, aes(x = dicamba_trt2, y = emmean2, color = strain_ID2)) +
  geom_point(size = 3, position = position_dodge(width = 0.3)) +
  geom_errorbar(aes(ymin = emmean2-se2, ymax = emmean2 + se2), width = 0, size = 1, position = position_dodge(width = 0.3)) +
  labs(x = "Herbicide Treatment", y = expression(paste("BNF (-1 \u00D7 Foliar",~ delta^{15},"N)")), color = "Strain") +
  theme_bw() +
  theme(axis.title = element_text(size = 18), axis.text = element_text(size = 16), legend.title = element_text(size = 18), legend.text = element_text(size = 16)) +
  geom_line(aes(group = strain_ID2), position = position_dodge(width = 0.3), size = 1) +
    scale_color_manual(values = c8)
N_rxn_norm 

### Analyze foliar delta-15N with nodule number as a covariate

N_per_nod_model <- lmer(delta_15N ~ dicamba_trt*strain_ID + dicamba_trt*plant_genotype + (1|Tray) + WK4_fungal_pathogen_minor + WK6_num_nodules, REML = FALSE, data = N_data4)
Anova(N_per_nod_model, type = 3) #sig effects of strain, fungal pathogen, and dic x strain, non sig effect of nodule number

simulationOutput <- simulateResiduals(fittedModel = N_per_nod_model, plot = F)
plot(simulationOutput) # passes

N_per_nod_emmeans <- emmeans(N_per_nod_model, pairwise ~ dicamba_trt|strain_ID, type = "response", adjust = "Tukey")
N_per_nod_emmeans 

N_per_nod_emmeans_dat <- as.data.frame(N_per_nod_emmeans$emmeans)

N_per_nod_emmeans_dat$strain_ID2 <- c("14479", "14479", "2063", "2063", "2087", "2087","2141", "2141", "2214", "2214", "2220", "2220", "2316", "2316", "4386", "4386")
N_per_nod_emmeans_dat$emmean2 <- N_per_nod_emmeans_dat$emmean*-1
N_per_nod_emmeans_dat$SE2 <- N_per_nod_emmeans_dat$SE*-1

N_per_nod_emmeans_dat$dicamba_trt2 <- "No Dicamba"
for(i in 1:nrow(N_per_nod_emmeans_dat)){
  if(N_per_nod_emmeans_dat[i,1] == 0.5){
    N_per_nod_emmeans_dat[i,11] <- "Dicamba"
  }
}

N_per_nod_emmeans_dat$dicamba_trt2 <- factor(N_per_nod_emmeans_dat$dicamba_trt2, levels = c("No Dicamba", "Dicamba"))

N_rxn_norm2 <- ggplot(N_per_nod_emmeans_dat, aes(x = dicamba_trt2, y = emmean2, color = strain_ID2)) +
  geom_point(size = 3, position = position_dodge(width = 0.3)) +
  geom_errorbar(aes(ymin = emmean2-SE2, ymax = emmean2 + SE2), width = 0, size = 1, position = position_dodge(width = 0.3)) +
  labs(x = "Herbicide Treatment", y = expression(paste("BNF (-1 \u00D7 Foliar",~ delta^{15},"N)")), color = "Strain") +
  theme_bw() +
  theme(axis.title = element_text(size = 18), axis.text = element_text(size = 16), legend.title = element_text(size = 18), legend.text = element_text(size = 16)) +
  geom_line(aes(group = strain_ID2), position = position_dodge(width = 0.3), size = 1) +
    scale_color_manual(values = c8)
N_rxn_norm2

### Spearman correlation: are strain rankings of foliar delta-15 N the same with or without dicamba?

N_emmeans_dat_dicamba <- N_emmeans_dat %>%
  filter(dicamba_trt == 0.5)
N_emmeans_dat_no_dicamba <- N_emmeans_dat %>%
  filter(dicamba_trt == 0)

cor.test(N_emmeans_dat_no_dicamba$emmean, N_emmeans_dat_dicamba$emmean, method = "spearman") # no significant correlation, S = 106, r = -.26 p = 0.54

```

## Plant size analysis
```{r}
#### This chunk analyzes plant size data (estimated from shoot biomass) from the microcosm study.

### Prep data

## Filter out uninoculated plants, missing data (e.g. plants from temporal block 2 that were not harvested for shoot biomass), and plants with severe symptoms from fungal pathogen contaminant
sb_dat <- plant_dat %>% filter(strain_ID != "uninoculated") %>%
  filter(WK4_fungal_pathogen_severe != "Y") %>%
  filter(shoot_biomass != "NA")

## Ensure variables are stored in correct class
sb_dat$Tray <- factor(sb_dat$Tray)
sb_dat$strain_ID <- factor(sb_dat$strain_ID)
sb_dat$dicamba_trt <- factor(sb_dat$dicamba_trt)
sb_dat$plant_genotype <- factor(sb_dat$plant_genotype)
sb_dat$shoot_biomass <- as.numeric(sb_dat$shoot_biomass)
sb_dat$WK4_fungal_pathogen_minor <- factor(sb_dat$WK4_fungal_pathogen_minor)

### Data exploration

## Visualize raw data
sb_plot <- ggerrorplot(sb_dat, x="strain_ID", y="shoot_biomass",  desc_stat="mean_se", color="dicamba_trt", position=position_dodge(.3), xlab="strain", ylab="plant size", add="jitter", add.params=list(color="dicamba_trt"), facet.by = "plant_genotype")
sb_plot + theme(axis.text.x = element_text(angle=90,hjust=1))

sb_plot2 <- ggerrorplot(sb_dat, x="strain_ID", y="shoot_biomass",  desc_stat="mean_se", color="plant_genotype", position=position_dodge(.3), xlab="strain", ylab="plant size", add="jitter", add.params=list(color="plant_genotype"), facet.by = "dicamba_trt")
sb_plot2 + theme(axis.text.x = element_text(angle=90,hjust=1))

## Examine distribution of data
plot(density(sb_dat$shoot_biomass)) # slightly right skewed
plot(density(log10(sb_dat$shoot_biomass))) # more normal
plot(density(sqrt(sb_dat$shoot_biomass))) # most normal

### Model Selection. We're choosing from mixed effects linear models to analyze square-root-transformed shoot biomass data because these models are appropriate for the data distribution (normal) and can evaluate random and fixed effect factors.

sb_lmer_mod1 <- lmer(sqrt(shoot_biomass) ~ strain_ID*dicamba_trt*plant_genotype + (1|Tray) + WK4_fungal_pathogen_minor, data = sb_dat, REML = FALSE)
AIC(sb_lmer_mod1) #239.9

sb_lmer_mod2 <- lmer(sqrt(shoot_biomass) ~ strain_ID*dicamba_trt*plant_genotype + (1|Tray), data = sb_dat, REML = FALSE)
AIC(sb_lmer_mod2) #236.2

sb_lmer_mod3 <- lmer(sqrt(shoot_biomass) ~ strain_ID*dicamba_trt + dicamba_trt*plant_genotype + strain_ID*plant_genotype + (1|Tray), data = sb_dat, REML = FALSE)
AIC(sb_lmer_mod3) #235.5

sb_lmer_mod4 <- lmer(sqrt(shoot_biomass) ~ strain_ID*dicamba_trt + dicamba_trt*plant_genotype + (1|Tray), data = sb_dat, REML = FALSE)
AIC(sb_lmer_mod4) #227.6 <- best-fit model

sb_models_all <- list(sb_lmer_mod1, sb_lmer_mod2, sb_lmer_mod3, sb_lmer_mod4)
aictab(sb_models_all)

### Model Validation
simulationOutput <- simulateResiduals(fittedModel = sb_lmer_mod4, plot = F)
plot(simulationOutput) # deviations in one quantile detected, combined adjustes quantile test n.s.
plot(sb_lmer_mod4) # no pattern detected
plot(histogram(residuals(sb_lmer_mod4))) # residulas are normally distributed

### Test hypotheses
Anova(sb_lmer_mod4, type = 3) # sig effects of dicamba and strain x dic

### Post-hoc test of interactions

## Calculate estimated marginal means from best-fitting model
sb_emmeans <- emmeans(sb_lmer_mod4, pairwise ~ dicamba_trt|strain_ID)

## Examine pairwise comparisons of dicamba effect by strain
sb_emmeans_contrast_dat <- as.data.frame(sb_emmeans$contrasts)
sb_emmeans_contrast_dat
summary(sb_emmeans, by = NULL, adjust = "dunnettx") # Dunnet p-value adjustments, these results are reported in Supplementary Table S3 (Part C)

### Visualize results

## Calculate back-transformed estimated marginal means for ease of interpretation when plotting
sb_emmeans2 <- emmeans(sb_lmer_mod4, pairwise ~ dicamba_trt|strain_ID, type = "response", adjust = "Tukey")
sb_emmeans2
sb_emmeans_dat <- as.data.frame(sb_emmeans2)
sb_emmeans_dat2 <- sb_emmeans_dat[1:16, ]

## Prep dataframe to generate plot
sb_emmeans_dat2$strain_ID2 <- c("14479", "14479", "2063", "2063", "2087", "2087","2141", "2141", "2214", "2214", "2220", "2220", "2316", "2316", "4386", "4386")

sb_emmeans_dat2$dicamba_trt2 <- "No Dicamba"
for(i in 1:nrow(sb_emmeans_dat2)){
  if(sb_emmeans_dat2[i,2] == 0.5){
   sb_emmeans_dat2[i,10] <- "Dicamba"
  }
}

sb_emmeans_dat2$dicamba_trt2 <- factor(sb_emmeans_dat2$dicamba_trt2, levels = c("No Dicamba", "Dicamba"))

## Plot results
sb_rxn_norm <- ggplot(sb_emmeans_dat2, aes(x = dicamba_trt2, y = response, color = strain_ID2)) +
  geom_point(size = 3, position = position_dodge(width = 0.3)) +
  geom_errorbar(aes(ymin = response-SE, ymax = response + SE), width = 0, size = 1, position = position_dodge(width = 0.3)) +
  labs(x = "Herbicide Treatment", y = "Plant Size (mg)", color = "Strain") +
  theme_bw() +
  theme(axis.title = element_text(size = 18), axis.text = element_text(size = 16), legend.title = element_text(size = 20), legend.text = element_text(size = 16)) +
  geom_line(aes(group = strain_ID2), position = position_dodge(width = 0.3), size = 1) +
    scale_color_manual(values = c8)
sb_rxn_norm

### Spearman correlation: are strain rankings of shoot biomass the same with or without dicamba?

sb_emmeans_dat_dicamba <- sb_emmeans_dat2 %>%
  filter(dicamba_trt == 0.5)
sb_emmeans_dat_no_dicamba <- sb_emmeans_dat2 %>%
  filter(dicamba_trt == 0)

cor.test(sb_emmeans_dat_no_dicamba$response, sb_emmeans_dat_dicamba$response, method = "spearman") # no significant correlation, S = 108, r = -.29 p = 0.50

### Analyze effect of rhizobial inoculation on plant size with and without dicamba

## Create dataframe of uninoculated plants
uninoc_dat <- plant_dat %>% 
  filter(WK4_fungal_pathogen_severe != "Y") %>%
  filter(shoot_biomass != "NA") %>%
  filter(strain_ID == "uninoculated")

uninoc_dat <- subset(uninoc_dat, select = c(Tray, strain_ID, dicamba_trt, plant_genotype, shoot_biomass))

## Subtract mean size of uninoculated plants from size of inoculated plants according to tray, plant genotype and herbicide treatment

inoc_dat <- sb_dat # dataset containing inoculated plants only
inoc_dat$WK6_rhizobial_effect <- 0.0

for(j in 1:nrow(uninoc_dat)){
  for(i in 1:nrow(inoc_dat)){
    if(inoc_dat[i,2]==uninoc_dat[j,1] && inoc_dat[i,4]==uninoc_dat[j,3] && inoc_dat[i,5] == uninoc_dat[j,4]){
      inoc_dat[i,15] <- inoc_dat[i,9] - uninoc_dat[j,5]
      i <- i + 1
    }
    else{
      i <- i + 1
    }
  }
  j <- j + 1
}

## T-test: is mean rhizobial effect (calculated at tray-level) on Week 6 plant size significantly different in the presence or absence of dicamba?

WK6_rhiz_eff_dat <- summarySE(data = inoc_dat, measurevar = "WK6_rhizobial_effect", groupvars = c("strain_ID", "dicamba_trt", "plant_genotype"))

WK6_rhiz_eff_dat_dicamba <- WK6_rhiz_eff_dat %>%
  filter(dicamba_trt == "0.5")
mean(WK6_rhiz_eff_dat_dicamba$WK6_rhizobial_effect) #-0.26

WK6_rhiz_eff_dat_no_dicamba <- WK6_rhiz_eff_dat %>%
  filter(dicamba_trt == "0")
mean(WK6_rhiz_eff_dat_no_dicamba$WK6_rhizobial_effect) #0.75

t.test(WK6_rhiz_eff_dat_no_dicamba$WK6_rhizobial_effect, WK6_rhiz_eff_dat_dicamba$WK6_rhizobial_effect, paired = TRUE, alternative = "two.sided")
# result: t = 3.8241, df = 15, p = 0.00166

## Visualize results
WK6_rhiz_eff_dat$strain_ID2 <- "strain"
for(i in 1:nrow(WK6_rhiz_eff_dat)){
  if(WK6_rhiz_eff_dat[i,1] == "strain_14479"){
    WK6_rhiz_eff_dat[i,9] <- "14479"
  }
  else if(WK6_rhiz_eff_dat[i,1] == "strain_2316"){
    WK6_rhiz_eff_dat[i,9] <- "2316"
  }
    else if(WK6_rhiz_eff_dat[i,1] == "strain_2220"){
    WK6_rhiz_eff_dat[i,9] <- "2220"
    }
    else if(WK6_rhiz_eff_dat[i,1] == "strain_2214"){
    WK6_rhiz_eff_dat[i,9] <- "2214"
    }
    else if(WK6_rhiz_eff_dat[i,1] == "strain_2141"){
    WK6_rhiz_eff_dat[i,9] <- "2141"
    }
    else if(WK6_rhiz_eff_dat[i,1] == "strain_2087"){
    WK6_rhiz_eff_dat[i,9] <- "2087"
    }
    else if(WK6_rhiz_eff_dat[i,1] == "strain_2063"){
    WK6_rhiz_eff_dat[i,9] <- "2063"
    }
    else if(WK6_rhiz_eff_dat[i,1] == "strain_4386"){
    WK6_rhiz_eff_dat[i,9] <- "4386"
    }
  i <- i + 1
}

WK6_rhiz_eff_dat$dicamba_trt2 = "Dicamba"

for(i in 1:nrow(WK6_rhiz_eff_dat)){
  if(WK6_rhiz_eff_dat[i,2] == "0"){
    WK6_rhiz_eff_dat[i,10] <- "No Dicamba"
  }
  i <- i + 1
}

WK6_rhiz_eff_dat$dicamba_trt2 <- factor(WK6_rhiz_eff_dat$dicamba_trt2, levels = c("No Dicamba", "Dicamba"))

WK6_rhiz_eff_plot <- ggplot(WK6_rhiz_eff_dat, aes(x = dicamba_trt2, y = WK6_rhizobial_effect, color = strain_ID2, shape = plant_genotype))+
  geom_point(size = 3, position = position_dodge(width = 0.3)) +
  geom_hline(yintercept = 0, linetype = "dashed") +
  labs(x = "Herbicide Treatment", y = "Rhizobial Effect on Plant Size (mg)", color = "Strain", shape = "Plant \nGenotype") +
  scale_color_manual(values = c8) +
  theme_classic() +
  theme(axis.title = element_text(size = 20), axis.text = element_text(size = 16), legend.text = element_text(size = 16), legend.title = element_text(size = 18))
WK6_rhiz_eff_plot

```

## Phylogeny-corrected analyses
```{r}
#### This chunk analyzes data from the microcosm study using models that are corrected to account for the phylogenetic relatedness of rhizobial strains. We compare model fit of phylogeny-corrected models with non phylogeny-corrected models with AIC.

### Phylogeny-corrected nodule number analysis
nod_num_phylo_dat <- nod_num_dat

## Make a phylogenetic tree where every tip is an individual plant (i.e. plate microcosm)
## First, check that all of the representative names on the tree appear in your data frame for indexing your sample names
name.check(rhizobia.tree, nod_num_phylo_dat, data.names = nod_num_phylo_dat$strain_ID) #OK


## Create "Plate_ID2" column so that every plate is labeled with a unique name across temporal blocks

nod_num_phylo_dat$Plate_ID2 <- "Plate"
nod_num_phylo_dat$Plate_ID2 <- paste0(nod_num_phylo_dat$Plate_ID2, seq_along(nod_num_phylo_dat$Plate_ID2))

## Change the sample ID to a character string so it can be added to the tree properly
nod_num_phylo_dat$Plate_ID2 <- as.character(nod_num_phylo_dat$Plate_ID2)

rhizobia.tree$tip.label # matches strain_ID column (column #3)

## Initialize the first tip graft. Bind tip adds new tips with a designated "tip.label" name
new.tree <- bind.tip(tree = rhizobia.tree,
                     tip.label = nod_num_phylo_dat[[16]][1], # sampleID reference - [[1]] gives the column [] gives the 
                     edge.length = 0, # the new tip has no length, it has the same relationship to other samples on the same tip
                     position = 0,
                     where = which(rhizobia.tree$tip.label==nod_num_phylo_dat[[3]][1])) # locate which existing tip to attach your new tip to. Use your tree$tip.labels and a matching column in your dataframe.

plot(new.tree) #Plate1 should appear where strain_2316 is
new.tree <- drop.tip(new.tree, "Plate1") # remove the first tip you tested from new.tree
plot(new.tree)

#rebuild new tree
new.tree <- bind.tip(tree = rhizobia.tree,
                     tip.label = nod_num_phylo_dat[[16]][1], # sampleID reference - [[1]] gives the column [] gives the 
                     edge.length = 0.0001, # the new tip has no length, it has the same relationship to other samples on the same tip
                     position = 0,
                     where = which(rhizobia.tree$tip.label==nod_num_phylo_dat[[3]][1]))

plot(new.tree)
## The loop below uses bind.tip to cycle through the data set and make a tip for each sample-ID
for (i in 2:nrow(nod_num_phylo_dat)) {
  new.tree <- bind.tip(tree = new.tree,
                       tip.label = nod_num_phylo_dat[[16]][i], # Tip labels are pulled from the
                       edge.length = 0.0001,
                       position = 0,
                       where = which(new.tree$tip.label==nod_num_phylo_dat[[3]][i]))
}

plot(new.tree)

## make an object of only the tip labels from the original tree
tip <- rhizobia.tree$tip.label

## remove the original representative branch tips from the newly generated tree (leaving only the samples)
nod_num_rhizobia_tree <- drop.tip(phy = new.tree, tip = tip, trim.internal = FALSE)

plot(nod_num_rhizobia_tree)

## Randomly resolve polytomies in the newly ammended tree, gives polytomies a randomly negligibly small distance from each other
nod_num_rhizobia_tree <- multi2di(nod_num_rhizobia_tree, random = T)
plot(nod_num_rhizobia_tree)

# Make a phylolm-ready dataframe by making sample IDs the rownames
nod_num_phylo_dat2 <- nod_num_phylo_dat
nod_num_phylo_dat2 <- column_to_rownames(nod_num_phylo_dat2, "Plate_ID2")

# Check that tips in tree and data in sb_phylo_dat3 match
name.check(nod_num_rhizobia_tree, nod_num_phylo_dat2) #OK

## Run Phylogenetically-corrected linear model for nodule number using pglmm using same formula as top model

nod_num_pglmm <- pglmm_compare(log(WK4_num_nodules) ~ dicamba_trt*plant_genotype + dicamba_trt*strain_ID + strain_ID*plant_genotype + (1|Tray) + WK4_fungal_pathogen_minor + temporal_block, family = "gaussian", phy = nod_num_rhizobia_tree, data = nod_num_phylo_dat2, REML = FALSE)
nod_num_pglmm #AIC is 543.7, Phylogenetic random effects variance (s2):  0.0000 

### Phylogeny-corrected BNF analysis
N_phylo_dat <- N_data4

## Make a phylogenetic tree where every tip is an individual plant (i.e. plate microcosm)
## First, check that all of the representative names on the tree appear in your data frame for indexing your sample names
name.check(rhizobia.tree, N_phylo_dat, data.names = N_phylo_dat$strain_ID) #OK

## Change the sample ID to a character string so it can be added to the tree properly
N_phylo_dat$Plate_ID <- as.character(N_phylo_dat$Plate_ID)

rhizobia.tree$tip.label # matches strain_ID column (column #4)

## Initialize the first tip graft. Bind tip adds new tips with a designated "tip.label" name
new.tree <- bind.tip(tree = rhizobia.tree,
                     tip.label = N_phylo_dat[[2]][1], # sampleID reference - [[1]] gives the column [] gives the 
                     edge.length = 0, # the new tip has no length, it has the same relationship to other samples on the same tip
                     position = 0,
                     where = which(rhizobia.tree$tip.label==N_phylo_dat[[4]][1])) # locate which existing tip to attach your new tip to. Use your tree$tip.labels and a matching column in your dataframe.

plot(new.tree) #ST1-A02 should appear where strain_2220 is
new.tree <- drop.tip(new.tree, "ST1-A02") # remove the first tip you tested from new.tree
plot(new.tree)

#rebuild new tree
new.tree <- bind.tip(tree = rhizobia.tree,
                     tip.label = N_phylo_dat[[2]][1], # sampleID reference - [[1]] gives the column [] gives the 
                     edge.length = 0.0001, # the new tip has no length, it has the same relationship to other samples on the same tip
                     position = 0,
                     where = which(rhizobia.tree$tip.label==N_phylo_dat[[4]][1]))

plot(new.tree)
## The loop below uses bind.tip to cycle through the data set and make a tip for each sample-ID
for (i in 2:nrow(N_phylo_dat)) {
  new.tree <- bind.tip(tree = new.tree,
                       tip.label = N_phylo_dat[[2]][i], # Tip labels are pulled from the
                       edge.length = 0.0001,
                       position = 0,
                       where = which(new.tree$tip.label==N_phylo_dat[[4]][i]))
}

plot(new.tree)

## make an object of only the tip labels from the original tree
tip <- rhizobia.tree$tip.label

## remove the original representative branch tips from the newly generated tree (leaving only the samples)
N_rhizobia_tree <- drop.tip(phy = new.tree, tip = tip, trim.internal = FALSE)

plot(N_rhizobia_tree)

## Randomly resolve polytomies in the newly ammended tree, gives polytomies a randomly negligibly small distance from each other
N_rhizobia_tree <- multi2di(N_rhizobia_tree, random = T)
plot(N_rhizobia_tree)

# Make a phylolm-ready dataframe by making sample IDs the rownames
N_phylo_dat2 <- N_phylo_dat
N_phylo_dat2 <- column_to_rownames(N_phylo_dat2, "Plate_ID")

# Check that tips in tree and data in sb_phylo_dat3 match
name.check(N_rhizobia_tree, N_phylo_dat2) #OK

## Run Phylogenetically-corrected linear model for N fixation using pglmm using same formula as top model

N_pglmm <- pglmm_compare(delta_15N ~ dicamba_trt*strain_ID + dicamba_trt*plant_genotype + (1|Tray) + WK4_fungal_pathogen_minor, family = "gaussian", phy = N_rhizobia_tree, data = N_phylo_dat2, REML = FALSE) #error: system is singular

### Phylogeny-corrected plant size analysis
sb_phylo_dat <- sb_dat

## Make a phylogenetic tree where every tip is an individual plant (i.e. plate microcosm)
## First, check that all of the representative names on the tree appear in your data frame for indexing your sample names
name.check(rhizobia.tree, sb_phylo_dat, data.names = sb_phylo_dat$strain_ID) #OK

rhizobia.tree$tip.label

## Change the sample ID to a character string so it can be added to the tree properly
sb_phylo_dat$Plate_ID <- as.character(sb_phylo_dat$Plate_ID)

## Initialize the first tip graft. Bind tip adds new tips with a designated "tip.label" name
new.tree <- bind.tip(tree = rhizobia.tree,
                     tip.label = sb_phylo_dat[[1]][1], # sampleID reference - [[1]] gives the column [] gives the 
                     edge.length = 0, # the new tip has no length, it has the same relationship to other samples on the same tip
                     position = 0,
                     where = which(rhizobia.tree$tip.label==sb_phylo_dat[[3]][1])) # locate which existing tip to attach your new tip to. Use your tree$tip.labels and a matching column in your dataframe.

plot(new.tree) #ST1-A01 should appear where strain_2316 is
new.tree <- drop.tip(new.tree, "ST1-A01") # remove the first tip you tested from new.tree
plot(new.tree)

#rebuild new tree
new.tree <- bind.tip(tree = rhizobia.tree,
                     tip.label = sb_phylo_dat[[1]][1], # sampleID reference - [[1]] gives the column [] gives the 
                     edge.length = 0.0001, # the new tip has no length, it has the same relationship to other samples on the same tip
                     position = 0,
                     where = which(rhizobia.tree$tip.label==sb_phylo_dat[[3]][1]))

plot(new.tree)
## The loop below uses bind.tip to cycle through the data set and make a tip for each sample-ID
for (i in 2:nrow(sb_phylo_dat)) {
  new.tree <- bind.tip(tree = new.tree,
                       tip.label = sb_phylo_dat[[1]][i], # Tip labels are pulled from the
                       edge.length = 0.0001,
                       position = 0,
                       where = which(new.tree$tip.label==sb_phylo_dat[[3]][i]))
}

plot(new.tree)

## make an object of only the tip labels from the original tree
tip <- rhizobia.tree$tip.label

## remove the original representative branch tips from the newly generated tree (leaving only the samples)
sb_rhizobia_tree <- drop.tip(phy = new.tree, tip = tip, trim.internal = FALSE)

plot(sb_rhizobia_tree)

## Randomly resolve polytomies in the newly ammended tree, gives polytomies a randomly negligibly small distance from each other
sb_rhizobia_tree <- multi2di(sb_rhizobia_tree, random = T)
plot(sb_rhizobia_tree)

# Make a phylolm-ready dataframe by making sample IDs the rownames
sb_phylo_dat2 <- sb_phylo_dat
sb_phylo_dat2 <- column_to_rownames(sb_phylo_dat2, "Plate_ID")

# Check that tips in tree and data in sb_phylo_dat3 match
name.check(sb_rhizobia_tree, sb_phylo_dat2) #OK

## Run Phylogenetically-corrected linear model for shoot biomass using pglmm

sb_pglmm <- pglmm_compare(sqrt(shoot_biomass) ~ dicamba_trt*strain_ID + dicamba_trt*plant_genotype + (1|Tray), family = "gaussian", phy = sb_rhizobia_tree, data = sb_phylo_dat2, REML = FALSE)
sb_pglmm #AIC is 229.3 #phylogenetic random effects variance (s2) = 0.00


```

## Manuscript figures
```{r}
#### Figure 1: time until nodulation results
fig1 <- ggarrange(nod_time_A, nod_timeB, ncol = 2, labels = c("A", "B"), font.label = list(size = 26, face = "bold"))
fig1

#### Figure 2: nodule number, BNF, and plant size results
all_rxn_norms <- ggarrange(nod_num_rxn_norm, N_rxn_norm, sb_rxn_norm, ncol = 3, nrow = 1, labels = c("A", "B", "C"), font.label = list(size = 22, face = "bold"), legend = c("top"), common.legend = TRUE)
all_rxn_norms

```
